libs/ledger-live-common/src/families/celo/api/sdk.ts

```diff
@@ -1,8 +1,93 @@
 import { ContractKit, newKit } from "@celo/contractkit";
 import { getEnv } from "../../../env";
+import { CeloVote } from "../types";
+import { makeLRUCache } from "../../../cache";
 
 let kit: ContractKit;
 export const celoKit = () => {
   if (!kit) kit = newKit(getEnv("API_CELO_NODE"));
   return kit;
 };
+
+/**
+ * Fetch account registered status. To lock any Celo, account needs to be registered first
+ */
+export const getAccountRegistrationStatus = async (address: string): Promise<boolean> => {
+  const accounts = await celoKit().contracts.getAccounts();
+  return await accounts.isAccount(address);
+};
+
+/**
+ * Fetch pending withdrawals, with an index
+ */
+export const getPendingWithdrawals = async (address: string) => {
+  const lockedGold = await celoKit().contracts.getLockedGold();
+  const pendingWithdrawals = await lockedGold.getPendingWithdrawals(address);
+  const pendingWithdrawalsWithIndexes = pendingWithdrawals
+    .map((withdrawal, index) => ({
+      ...withdrawal,
+      index,
+    }))
+    .sort((a, b) => a.time.minus(b.time).toNumber());
+  return pendingWithdrawalsWithIndexes;
+};
+
+/**
+ * Fetch all votes
+ */
+export const getVotes = async (address: string): Promise<CeloVote[]> => {
+  const election = await celoKit().contracts.getElection();
+  const voter = await election.getVoter(await voteSignerAccount(address));
+  const activates = await getActivateTransactionObjects(address);
+  const activatableValidatorGroups = activates.map(
+    (activate) => activate.txo.arguments[0]
+  );
+
+  const votes: CeloVote[] = [];
+  voter.votes.forEach((vote) => {
+    let activeVoteRevokable = true;
+    if (vote.pending.gt(0)) {
+      // If there's a pending vote, it has to be revoked first
+      activeVoteRevokable = false;
+      votes.push({
+        validatorGroup: vote.group,
+        amount: vote.pending,
+        // Not all pending votes can be activated, 24h has to pass
+        activatable: activatableValidatorGroups.includes(vote.group),
+        revokable: true,
+        index: 0,
+        type: "pending",
+      });
+    }
+    if (vote.active.gt(0))
+      votes.push({
+        validatorGroup: vote.group,
+        amount: vote.active,
+        activatable: false,
+        revokable: activeVoteRevokable,
+        index: 1,
+        type: "active",
+      });
+  });
```

Comment:
☝️ Here, you can condense the code a bit by doing something like:

```
 const votes: CeloVote[] = voter.votes.map((vote) => {
    const isActive = !vote.pending.gt(0); && vote.active.gt(0);

    return {
      validatorGroup: vote.group,
      amount: isActive ? vote.active : vote.pending,
      activatable: false,
      revokable: isActive
        ? true
        : // Not all pending votes can be activated, 24h has to pass
          activatableValidatorGroups.includes(vote.group),
      index: 1,
      type: isActive ? "active" : "pending",
    };
  });
```

Reply:
good idea, I'll do that

Reply:
so I coded it this way because the structure we get from the node looks like this:
```
export interface GroupVote {
    group: Address;
    pending: BigNumber;
    active: BigNumber;
}
```
so every vote consists of `pending` and `active` amount. And structure we want to use is a vote per pending/active, so for every `voter.votes` we create 1 or 2 votes

Reply:
Ah ok - I was wondering if that was a scenario we needed to support.  Looks good!